ci/script.sh

```diff
@@ -48,7 +48,7 @@ case $TARGET in
     ;;
 
   clippy)
-    cargo clippy
+    cargo clippy --workspace -- -D warnings -A clippy::unnecessary-cast
```

Comment:
The allow here is because the benchmarking script spit out bad code when a lot of our weights were generated. This was fixed in https://github.com/paritytech/substrate/commit/c29095026871c2b7d42734226dd45db8051d258e, but I didn't want to regenerate all the benches as part of this PR.

I've filed #1201 to track this separately